### ISSUE TITLE
Session closed. Most likely the page has been closed

### ISSUE THREAD

**Issue by RyanRyan99** [pedroslopez/whatsapp-web.js#1667]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the bug

After a while connected, and want to send a message I get a message like this

Error: Protocol error (Runtime.callFunctionOn): Session closed. Most likely the page has been closed.

### Expected behavior

Whatsapp stays connected and doesn't have to be restarted

### Steps to Reproduce the Bug or Issue

1. Connected Account
2. Wait for a moment
3. Trying to send a message
4. Error

### Browser Type

Chromium

### WhatsApp Account Type

Standard

### Does your WhatsApp account have multidevice enabled?

Yes, I am using Multi Device

### Environment

Windows OS
whatsapp-web.js 1.17.1 version

**Comment by albanirneves** [comment]
Same problem here

**Comment by lucacasanova** [comment]
Same problem here.

**Comment by chagasjoao** [comment]
Same problem here.

**Comment by kulibinRuslan** [comment]
I also have this problem.

**Comment by scisammy** [comment]
Same problem here.

**Comment by jmarcosltc** [comment]
Same problem here. It appears only after I try to autheticate a new session after closing the last one, without restarting the code.

**Comment by albanirneves** [comment]
any news about it?

**Comment by Useems** [comment]
> Same problem here. It appears only after I try to autheticate a new session after closing the last one, without restarting the code.

**Comment by het-t** [comment]
facing the same issue...

**Comment by Judahh** [comment]
For me it appears if the computer locks. (Mac M1)

**Comment by siteng98** [comment]
Same issue

**Comment by pc-robelbois** [comment]
+1

**Comment by ivangrosse** [comment]
Same problem here

**Comment by onurguven** [comment]
Same problem here.

**Comment by pc-robelbois** [comment]
> Any workaround for this issue?

Clear session, make login again (no prod-ready walkaround)

**Comment by AwesomeAbhaySingh** [comment]
Please tell me solution for this problem

**Comment by alechkos** [comment]
Duplicate of #739

**Comment by KaranGhorse** [comment]
I have created a function which continues to perform some or the other activities from WhatsApp client in the inactive state of the user in which the session should not be closed but still the session is being closed whyyyy

**Comment by salah55s** [comment]
add this code, it worked for me


client.on('ready', () => {
    console.log('WhatsApp Client is ready!');
    startKeepAlive(); // Start the keep-alive mechanism here
});


// Keep-alive mechanism: Simulate typing
async function startKeepAlive() {
    const keepAliveInterval = 1 * 60 * 1000; // 1 minute ( change it and tell us what is the best use for that) 

    setInterval(async () => {
        try {
            console.log('Simulating typing activity...');
            const chat = await client.getChatById("201557421703@c.us"); // Or a test chat
            if (chat) {
                await chat.sendStateTyping();
                await new Promise(resolve => setTimeout(resolve, 2000)); // Simulate for 2 seconds
                await chat.clearState();
                console.log('Typing activity simulated.');
            }
        } catch (error) {
            console.error('Error simulating typing:', error.message);
        }
    }, keepAliveInterval);
}

**Comment by 2017jw** [comment]
set handleSIGHUP to false in puppeteer config because puppeteer's default config is true, when run whatsappwebjs as service and close your ssh session, sighup is sent to puppeteer, then chrome is closed when handleSIGHUP is set true.
